comfy_extras/nodes_clip_sdxl.py

```diff
@@ -48,7 +70,38 @@ def encode(self, clip, width, height, crop_w, crop_h, target_width, target_heigh
             while len(tokens["l"]) > len(tokens["g"]):
                 tokens["g"] += empty["g"]
         cond, pooled = clip.encode_from_tokens(tokens, return_pooled=True)
-        return ([[cond, {"pooled_output": pooled, "width": width, "height": height, "crop_w": crop_w, "crop_h": crop_h, "target_width": target_width, "target_height": target_height}]], )
+        return ([[
+            cond,
+            {
+                "pooled_output": pooled,
+                "width": width, "height": height,
+                "crop_w": crop_w, "crop_h": crop_h,
+                "target_width": target_width, "target_height": target_height
+            }
+        ]], )
+
+    @classmethod
+    def IS_CHANGED(cls, clip, width, height, crop_w, crop_h, target_width, target_height, text_g, text_l):
+        text_g_changed = is_dynamic_prompt_changed(text_g)
+        text_l_changed = is_dynamic_prompt_changed(text_l)
+        # If either is NaN, return NaN, otherwise return the sum
+        if math.isnan(text_g_changed) or math.isnan(text_l_changed):
+            return float("nan")
+        else:
+            # Log to console
+            return text_g_changed + text_l_changed
+
+    # @classmethod
+    # def VALIDATE_INPUTS(cls, clip, width, height, crop_w, crop_h, target_width, target_height, text_g, text_l):
+    #     # These return `True`, or an error message if the input is invalid
+    #     text_g_valid = validate_dynamic_prompt(text_g)
+    #     text_l_valid = validate_dynamic_prompt(text_l)
+    #     # If either is not `True`, return the error message, otherwise return `True`
+    #     if text_g_valid is not True:
+    #         return text_g_valid
+    #     elif text_l_valid is not True:
+    #         return text_l_valid
```

Comment:
There's a variety of commented out and unrelated changes in this PR - I'm not sure what Comfy's opinion on the PR itself is, but if he wants it, unrelated changes would need to be cleaned out beforehand